### PR TITLE
[Python] fix model to dict

### PR DESCRIPTION
**Pull request by JoeCqupt**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by JoeCqupt** [comment]
the item of list maybe dict type. so we should handle dict specially. 

call for review @spacether

File: modules/openapi-generator/src/main/resources/python/model_utils.mustache [hunk 1]
```diff
@@ -1369,14 +1370,17 @@ def model_to_dict(model_instance, serialize=True):
                            res.append(v)
                        elif isinstance(v, ModelSimple):
                            res.append(v.value)
+                       elif isinstance(v, dict):
+                           res.append(dict(map(
+                               extract_item,
+                               v.items()
+                           )))
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python/model_utils.mustache:1377]: Can you please include tests?
It's not clear to me how this code can take the mapped tuple of (key, value) and generating a value based on value only can then make a dict of that data.
Tests will verify that this is working correctly for the dict and list use cases.

Comment (JoeCqupt) [reply modules/openapi-generator/src/main/resources/python/model_utils.mustache:1377]: those code copy from line:1381 . i only do define a var for lambda expression.

**Review by spacether — APPROVED** [review]
Looks great; thank you for your PR and adding the tests!

**Pull request merged**